docs/ha_3scale.adoc

```diff
@@ -217,4 +217,67 @@ In this setup, to ensure good database access latency, each cluster will have it
+
+This is an example flow since a client wants a *System* new dev portal and incoming traffic can eventually reach system-app pods thanks to a newly created OpenShift Route:
+
+. System-app receives the request from UI/API to create a dev/admin portal or apicast OpenShift  Route
+. System-app enqueues the background job to create those OpenShift Routes into system-redis
+. System-sidekiq takes the job from system-redis, process it and contacts with zync API (zync deployment)
+. Zync API (zync deployment) creates a background job into zync-database (postgres)
+. Zync-que takes the route-creation-job from zync-database (postgres) and create the final OpenShift  Routes into the cluster that proccessed that job
+
+When having N clusters, imagine Cluster-A and Cluster-B, any of them can receive traffic managed by either Cluster-A-system-app or Cluster-B-system-app, so they will enqueue the job into its own database, and would follow the whole procedure:
+
+```
+System-app -> system-redis -> system-sidekiq -> zync (api) -> zync-database -> zync-que -> OpenShift Routes creation
+```
+
+And finally, the OpenShift Routes will be created *only* into the cluster that processed the route-creation-job (not on both OCP clusters), so incoming traffic managed by OpenShift Router will work only on one cluster (the one with the needed OpenShift Routes), so not having *active*/*active*.
+
+____
+*NOTE*
+
+*This application limitation is a hard limitation because the first worker that fetches the job, will create the OpenShift Route on the cluster it's running. So, each cluster will have a subset of OpenShift Routes.*
+____
+
+=== Active independent clusters
+
+A few customers wanting *active*/*active*, as it can not be achieved within the same 3scale instance due to the hard limitation imposed by the OpenShift Routes created by *Zync*, what they have done is to deploy 3scale independent instances on different OCP clusters, where every 3scale instance has its own databases, which are not shared and not synced between them.
+
+Here you can find some considerations:
+
+- Each 3scale independent instance will need its own High Availability configuration
+- You need to manage the 3scale configuration with a shared CI/CD, to ensure that any independent 3scale instance will have exactly the same configuration
+- You need a kind of Global Load Balancer (maybe done at DNS level above all OCP clusters), to ensure that the same % of traffic goes to every independent 3scale instance on its OCP cluster (maybe using round-robin policy)
+- The rate limit will be independent for every 3scale instance, so if for example you have 2 3scale independent instances, and want a global rate limit of 100 requests/minute, you will need to configure a rate limit of 50 requests/minute on each 3scale instance, and in case of having round-robin DNS policy on the Global Load Balancer, *theoretically* you will achieve the supposed global 100 requests/minute rate limit
```

Comment:
Wouldn't there also be the consideration if RH SSO is integrated for OIDC on the API traffic then duplication of clients across the multiple 3scale instances (so the same token can be authorised in any 3scale) would then cause synchronisation issues and ultimately data inconsistencies? Seems like another zync hard limitation maybe?

Reply:
Mmmm interesting, I have no experience about using 3scale with RH SSO integration (on 3cale SaaS we dont provide our own RH SSO, any customer can use their own OIDC solution that will be contacted by Zync-SaaS).

I would like to know more details about this specific possible problem you have seen with 3scale customers and weird HA solutions

Can you please share a bit more detail about it?

Reply:
Hey @kevprice83 , did you have time to think about the possibility of shared RH SSO between independent 3scale instances (using same or different wildcard domain)??

Reply:
@slopezz so thinking about the proposal here: independent active 3scale instances not sharing databases, the issue to be careful with when integrating with RH SSO is if the RH SSO is shared between 3scale instances then the shared CI/CD should probably solve that problem. The shared CI/CD would ensure that the state of all objects synchronised between 3scale and RH SSO, via zync, is consistent. 

Regarding the `WILDCARD_DOMAIN` this would not impact the RH SSO integration to my knowledge. Only if RH SSO was not shared (1 per 3scale instance) would they need to take care of the DNS on the RH SSO side so that it was consistent which would then ensure the `iss` claim is always the same and therefore it won't matter which 3scale instance the API request + JWT reaches as it would be valid in both.

I would also expect that in this scenario you are describing the `WILDCARD_DOMAIN` is always the same and the DNS + loadbalancing is handled on the OpenShift networking layer.

Does that sound like it fits with the architecture you had in mind here?

Reply:
Thanks @kevprice83 for checking it, that's what I had in mind

I have updated a bit the case of possible active independent clusters on commit https://github.com/3scale/3scale-Operations/pull/17/commits/09e5b022594118c011993e2731f76d8aa7ce184c, please check it, if you think it is good enough, please approve the PR and I will merge it